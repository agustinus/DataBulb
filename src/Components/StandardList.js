import React from 'react';
import {View, FlatList, StyleSheet} from 'react-native';
import PropTypes from 'prop-types';

export default class StandardList extends React.PureComponent {
  static propTypes = {
    renderItem: PropTypes.func.isRequired,
    data: PropTypes.oneOfType([PropTypes.object, PropTypes.array]).isRequired,
    onRefresh: PropTypes.func,
    isRefreshing: PropTypes.bool,
    style: PropTypes.object,
    onEndReached: PropTypes.func,
    onScroll: PropTypes.func,
  };

  static defaultProps = {
    isRefreshing: false,
  };

  _keyExtractor = (item, index) => index.toString();

  constructor(props) {
    super(props);
  }

  _itemSeparator = () => {
    return (
      //Item Separator
      <View style={styles.listSeparator} />
    );
  };

  render() {
    return (
      <FlatList
        style={this.props.style}
        data={this.props.data}
        extraData={this.state}
        keyExtractor={this._keyExtractor}
        renderItem={({item}) => this.props.renderItem({item})}
        ItemSeparatorComponent={this._itemSeparator}
        onRefresh={this.props.onRefresh}
        refreshing={this.props.isRefreshing}
        onEndReached={this.props.onEndReached}
        onEndReachedThreshold={0.3}
        onScroll={this.props.onScroll}
      />
    );
  }
}

const styles = StyleSheet.create({
  listSeparator: {
    height: 10,
    width: '100%',
  },
  textBox: {
    backgroundColor: 'black',
    color: 'white',
    width: 40,
    textAlign: 'center',
    fontWeight: 'bold',
  },
});
