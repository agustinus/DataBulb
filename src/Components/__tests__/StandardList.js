import React from 'react';
import StandardList from '../StandardLIst';

import renderer from 'react-test-renderer';

describe('StandardList', () => {
  let tree = null;
  let instance = null;

  beforeAll(() => {
    tree = renderer.create(
      <StandardList
        renderItem={item => {
          return 'test item';
        }}
        data={[{title: 'test title'}]}
      />,
    );
    instance = tree.getInstance();
  });

  test('renders correctly', () => {
    expect(tree.toJSON()).toMatchSnapshot();
  });

  test('function: keyExtractor converts index to string', () => {
    expect(instance._keyExtractor(null, 1)).toMatch('1');
  });

  test('function: itemSeparator renders the correct style', () => {
    expect(instance._itemSeparator()).toMatchSnapshot();
  });

  test('renders items correctly', () => {
    expect(instance.props.renderItem()).toMatchSnapshot();
  });
});
