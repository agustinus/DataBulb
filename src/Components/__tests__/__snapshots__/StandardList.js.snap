// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StandardList function: itemSeparator renders the correct style 1`] = `
<View
  style={
    Object {
      "height": 10,
      "width": "100%",
    }
  }
/>
`;

exports[`StandardList renders correctly 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  data={
    Array [
      Object {
        "title": "test title",
      },
    ]
  }
  disableVirtualization={false}
  extraData={null}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={0.3}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshing={false}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      test item
    </View>
  </View>
</RCTScrollView>
`;

exports[`StandardList renders items correctly 1`] = `"test item"`;
